addon/components/nrg-render-template-block.js

```diff
@@ -1,59 +1,42 @@
-import Component from '@ember/component';
-import layout from '../templates/components/nrg-render-template-block';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
 
-export default Component.extend({
-  layout,
-  classNames: ['is-visually-hidden'],
-
-  getTemplateNode() {
-    return this.element.querySelector('.js-capture-node');
-  },
-
-  setupMutationObserver() {
-    const callback = () => this.updateParent();
-    const node = this.getTemplateNode();
-    const config = {
-      attributes: true,
-      childList: true,
-      subtree: true,
-    };
-
-    this.mutationObserver = new MutationObserver(callback);
-    this.mutationObserver.observe(node, config);
-  },
-
-  teardownMutationObserver() {
-    if (this.mutationObserver) {
-      this.mutationObserver.disconnect();
-    }
-  },
+export default class NrgRenderTemplateBlockComponent extends Component {
+  mutationObserver;
+  captureElement;
 
   removeIdsInDomTree(node) {
     node.id = '';
     const children = node.children || [];
     for (let i = 0; i < children.length; i++) {
       this.removeIdsInDomTree(children[i]);
     }
-  },
+  }
 
   updateParent() {
-    const clonedNode = this.getTemplateNode().cloneNode(true);
+    const clonedNode = this.captureElement.cloneNode(true);
     this.removeIdsInDomTree(clonedNode);
-    this.templateRendered && this.templateRendered(clonedNode);
-  },
+    this.args.templateRendered?.(clonedNode);
+  }
 
-  didInsertElement() {
-    this._super(...arguments);
-    this.setupMutationObserver();
-  },
+  @action
+  setupMutationObserver(element) {
+    this.captureElement = element;
+    const callback = () => this.updateParent();
```

Comment:
Do you need a function to call a function? Or can you just use
```suggestion
    const callback = this.updateParent;
```

Reply:
In this case, I'm passing the function as a callback to a native javascript construct.
This means that `this` will be the mutation observer when `updateParent` is called.
I want `this` to be the component, so I wrap it in an arrow function.